### ISSUE TITLE
Factor out integration tests from `vehicle-python`

### ISSUE THREAD

**Issue by wenkokke** [vehicle-lang/vehicle#268]
The tests for `vehicle-python` rely on `vehicle` to transform `.vcl` files to `.json`.

These tests should be rewritten to take the `.json` files as input, so that we don't need `vehicle` to run the tests.

Having these separate would allow the CI to move much faster—currently, even running the tests for `vehicle-python` requires the entire test suite for `vehicle` to pass.

**Comment by MatthewDaggitt** [comment]
Hmm we used to do this and then I encouraged @Tgl70 to switch to calling Vehicle directly. My reasoning was that it was going to be very laborious to recreate the JSON every time @ndslusarz makes a tweak to the JSON format?

How can we avoid having to regenerate the JSON for every test if it changes?

**Comment by wenkokke** [comment]
Update them on commit? That way you need Vehicle locally, but the CI can move fast?

**Comment by wenkokke** [comment]
Alternatively, have the CI generate the JSON and fail & open a PR if they differ?

**Comment by Yiergot** [comment]
Wouldn't it be best to create a JSON schema and use that to test?

**Comment by MatthewDaggitt** [comment]
Ooh can you automatically generate queries from a JSON schema?

**Comment by wenkokke** [comment]
No, but you can check conformity,  generate the entire Python dataclasss structure from it & load data into that dataclass structure from it, & get editor support with autocompletion and documentation.

Though if there isn't a JSON schema quickcheck module we could probably write one!

**Comment by MatthewDaggitt** [comment]
That all seems useful.

Another random idea thrown out. Could we use the golden json files from the compiler tests? Or are you against that coupling @wenkokke ?

**Comment by wenkokke** [comment]
That would result in trouble when packaging `vehicle-python` since those files wouldn't be included.

Might be able to do this using symlinks, if Poetry picks those up?

Unless we don't package the test code with the source distributions?

**Comment by MatthewDaggitt** [comment]
The `vehicle-python` package now exposes a lot more of Vehicle than simply JSON parsing so I don't think it's feasible to decouple the Python tests from the Haskell compiler in this way. Closing.
